test/LineTest.cpp

```diff
@@ -273,14 +271,59 @@ TEST(LineTests, IntersectionsDifferentTypes) {
     EXPECT_TRUE(A1.doesIntersect(AS2));
     EXPECT_EQ(*A1.intersects(AS2), Vector2(4.0, 0.0));
 
-
-//edge cases
-    Vector2 R5(1.0, 3.0), R6(1.0, - 1.0), R7(2.0, 0.0), R8(3.0, 0.0);
+    // edge cases
+    Vector2 R5(1.0, 3.0), R6(1.0, -1.0), R7(2.0, 0.0), R8(3.0, 0.0);
     Line A3(R5, R6), A4(R2, R7), A5(R7, R8);
     LineSegment AS3(R5, R6), AS4(R2, R7), AS5(R7, R8);
 
     ASSERT_NE(AS1.intersects(AS4), std::nullopt);
     EXPECT_TRUE(AS1.doesIntersect(AS4));
     EXPECT_EQ(*AS1.intersects(AS4), R2);
+}
+
+TEST(LineTests, ForwardIntersect) {
+    Vector2 forwardLine_start(0.0, 0.0), forwardLine_end(2.0, 2.0);
+    Vector2 line1_start(-1.0, -3.0), line1_end(-1.0, -2.0), line2_start(1.0, 4.0), line2_end(1.0, 5.0), line3_start(-1.0, 1.0), line3_end(1.0, -1.0),
+        line4_start(1000000000.0, -1000000000.0), line4_end(1000000000.0, -99999999.0), line5_start(1.0, 1.0), line5_end(3.0, 3.0);
+    Line forwardLine(forwardLine_start, forwardLine_end), line1(line1_start, line1_end), line2(line2_start, line2_end), line3(line3_start, line3_end),
+        line4(line4_start, line4_end), line5(line5_start, line5_end);
+    std::optional<Vector2> result = line1.forwardIntersect(forwardLine);
+    EXPECT_FALSE(result.has_value());  // They do intersect, but not in positive direction
+    result = line2.forwardIntersect(forwardLine);
+    EXPECT_EQ(result, Vector2(1.0, 1.0));  // An half-way past intersection in positive direction
+    result = line3.forwardIntersect(forwardLine);
+    EXPECT_EQ(result, Vector2(0.0, 0.0));  // Boundary case where an intersection happens at the start of the forwardline
+    result = line4.forwardIntersect(forwardLine);
+    EXPECT_EQ(result, Vector2(1000000000.0, 1000000000.0));  // Far away forward intersection for both lines
+    result = line5.forwardIntersect(forwardLine);
+    EXPECT_FALSE(result.has_value());  // No intersection is returned in case the lines are the same
+}
 
+TEST(LineTests, Shadow) {
+    Vector2 source(1.0, 2.0), projectLine_start(-4.0, -6.0), projectLine_end(-4.0, 6.0), blockLine1_start(0.0, 2.0), blockLine1_end(0.0, 2.0), blockLine2_start(-1.0, 0.0),
+        blockLine2_end(-1.0, 2.0), blockLine3_start(1.0, 1.0), blockLine3_end(0.0, 0.0), blockLine4_start(1.0, 3.0), blockLine4_end(0.0, 2.0), blockLine5_start(0.0, -4.0),
+        blockLine5_end(0.0, 4.0), blockLine6_start(1.0, 2.0), blockLine6_end(1.0, 2.0), blockLine7_start(1.0, 0.0), blockLine7_end(1000000000.0, 0.0), blockLine8_start(-1.0, 1.0),
+        blockLine8_end(-3.0, -1.0), checkLine2_start(-4.0, -3.0), checkLine2_end(-4.0, 2.0), checkLine4_start(-4.0, 2.0), checkLine4_end(-4.0, 6.0), checkLine8_start(-4.0, -0.5),
+        checkLine8_end(-4.0, -1.75);
```

Comment:
Agreed with Rolf. This makes everything a lot more readable (and I did it in 30 seconds by pressing enter quickly
```suggestion
    Vector2 source(1.0, 2.0),
    projectLine_start(-4.0, -6.0), 
    projectLine_end(-4.0, 6.0), 
    blockLine1_start(0.0, 2.0), 
    blockLine1_end(0.0, 2.0), 
    blockLine2_start(-1.0, 0.0),
    blockLine2_end(-1.0, 2.0), 
    blockLine3_start(1.0, 1.0), 
    blockLine3_end(0.0, 0.0), 
    blockLine4_start(1.0, 3.0), 
    blockLine4_end(0.0, 2.0), 
    blockLine5_start(0.0, -4.0),
        blockLine5_end(0.0, 4.0), 
        blockLine6_start(1.0, 2.0), 
        blockLine6_end(1.0, 2.0), 
        blockLine7_start(1.0, 0.0), 
        blockLine7_end(1000000000.0, 0.0), 
        blockLine8_start(-1.0, 1.0),
        blockLine8_end(-3.0, -1.0), 
        checkLine2_start(-4.0, -3.0), 
        checkLine2_end(-4.0, 2.0), 
        checkLine4_start(-4.0, 2.0), 
        checkLine4_end(-4.0, 6.0), 
        checkLine8_start(-4.0, -0.5),
        checkLine8_end(-4.0, -1.75);
```

Reply:
Well it is not in style with the other test cases